src/CMakeLists.txt

```diff
@@ -17,9 +15,31 @@
 include(ExternalProject)
 
 find_package(Threads REQUIRED)
+set(CMAKE_THREAD_PREFER_PTHREAD TRUE)
+set(THREADS_PREFER_PTHREAD_FLAG TRUE)
+
+set(OPENSSL_USE_STATIC_LIBS TRUE)
 find_package(OpenSSL REQUIRED)
-find_package(Readline REQUIRED)
 
+# Create compile_commands.json
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+
+if (${CMAKE_SYSTEM_NAME} MATCHES "Windows")
+  set(MGCONSOLE_ON_WINDOWS TRUE)
+elseif(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
+  set(MGCONSOLE_ON_LINUX TRUE)
+elseif(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+  set(MGCONSOLE_ON_OSX TRUE)
+endif()
```

Comment:
First, prefer STREQUAL.
Second, for this use case, use 
https://cmake.org/cmake/help/latest/manual/cmake-variables.7.html#variables-that-describe-the-system
(WIN32, UNIX...)

Reply:
Sure.